src/swap/components/SwapProvider.tsx

```diff
@@ -100,9 +101,28 @@ export function SwapProvider({
     lifeCycleStatus,
     lifeCycleStatus.statusData, // Keep statusData, so that the effect runs when it changes
     lifeCycleStatus.statusName, // Keep statusName, so that the effect runs when it changes
-    resetInputs,
   ]);
 
+  useEffect(() => {
+    // Reset inputs after status reset
+    if (lifeCycleStatus.statusName === 'init' && hasHandledSuccess) {
+      setHasHandledSuccess(false);
+      resetInputs();
+    }
+  }, [hasHandledSuccess, lifeCycleStatus.statusName, resetInputs]);
+
+  useEffect(() => {
+    // Reset status to init after success has been handled
+    if (lifeCycleStatus.statusName === 'success' && hasHandledSuccess) {
+      setLifeCycleStatus({
+        statusName: 'init',
+        statusData: {
+          isMissingRequiredField: false,
+        },
+      });
+    }
+  }, [hasHandledSuccess, lifeCycleStatus.statusName]);
```

Comment:
Can we add some tests to cover these cases please.

Reply:
added

Reply:
Just sync with @abcrane123 , she will take a final pass on this PR, and than we can get it approved.